internal/cli/opsmanager/serverusage/download.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverusage
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+type DownloadOpts struct {
+	cli.DownloaderOpts
+	startDate string
+	endDate   string
+	format    string
+	store     store.ServerUsageReportDownloader
+}
+
+var downloadMessage = "Download of %s completed.\n"
+
+func (opts *DownloadOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DownloadOpts) Run() error {
+	f, err := opts.NewWriteCloser()
+	if err != nil {
+		return err
+	}
+
+	if err := opts.store.DownloadServerUsageReport(opts.newServerTypeOptions(), f); err != nil {
+		_ = opts.OnError(f)
+		return err
+	}
+	if opts.Out != "/dev/stdout" {
+		fmt.Printf(downloadMessage, opts.Out)
+	}
+	return f.Close()
+}
+
+func (opts *DownloadOpts) newServerTypeOptions() *opsmngr.ServerTypeOptions {
+	return &opsmngr.ServerTypeOptions{
+		StartDate:  opts.startDate,
+		EndDate:    opts.endDate,
+		FileFormat: opts.format,
+	}
+}
+
+func (opts *DownloadOpts) initDefaultOut() {
+	opts.format = "." + opts.format
+	if opts.Out == "" {
+		opts.Out = "serverUsage-" + opts.startDate + "-" + opts.endDate + opts.format
```

Comment:
I think you're missing a dot(.) here, opts.endDate + "." + opts.format
Also optional, using `fmt.Sprintf` could make this a bit more readable

Reply:
I don't because I am inserting `.` at line 71